docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,228 @@
+
+* allowing long timeout commit discussed here: https://github.com/tendermint/tendermint/issues/5911#issuecomment-973560381
+
+## Decision
+
+None
+
+## Detailed Design
+
+### New Consensus Parameters
+
+A new `TimeoutParams` `message` will be added to the [params.proto file][consensus-params-proto].
+This message will have the following form:
+
+```proto
+message TimeoutParams {
+ google.protobuf.Duration timeout_propose = 1
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_propose_delta = 2
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote = 3
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote_delta = 4
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_commit = 5
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ bool skip_timeout_commit = 6;
+}
+```
+
+This new `message` will then be added as a field into the [`ConsensusParams`
+message][consensus-params-proto]. The same default values that are [currently
+set for these parameters][current-timeout-defaults] in the local configuration
+file will be used as the defaults for these new consensus parameters in the
+[consensus parameter defaults][default-consensus-params].
+
+Validation of these new parameters will be identical to their [validation
+currently][time-param-validation].
+Namely, these parameters must be non-negative.
+
+### Removal of Old Parameters
+
+The old `timeout-*` parameters that are configured in the [config.toml][config-toml]
+will be removed completely.
+
+### Optional: Temporary Local Overrides
+
+The new `TimeoutParams` will be released during the `v0.36` release cycle. Accidentally
+configuring these parameters too low would result could result in chains with slow
+networks or low-degree network topologies to not gossip votes within the configured
+timeouts and require many rounds for consensus to occur. To prevent this condition,
+we could optionally include a set of `unsafe-timeout-*` parameters in the `config.toml`.
+
+These parameters would allow node operators to quickly remedy the situation while
+preparing to update the consensus parameters. This would be a temporary solution that
+we would remove within the v0.37 release.
+
+### Add New Consensus Parameters to HashedParams
+
+Tendermint currently only verifies that a subset of the consensus parameters are
+equal across all validators. These parameters are the `BlockMaxBytes` and the `BlockMaxGas`.
+A [hash of these parameters][hashed-params] is included in the block. Validators ensure 
+their values of the parameters match by hashing their value of the parameters and 
+checking that their hashed value matches the hash included in the block. 
+
+Including the new parameters in the consensus parameters hash will break verification
```

Comment:
Could we finesse this by only hashing non-default values?

Reply:
I think the only gotcha here is that it means "the defaults at the time of the initial upgrade". We'd have to pin those against other changes in the defaults we might make later.

Reply:
That's a crucial caveat. 

updated the doc to reflect this upgrade scheme.